### ISSUE TITLE
Using state variable inside reaction always return undefined

### ISSUE THREAD

**Issue by MatthieuCoelho** [mobxjs/mobx-react-lite#129]
I'm creating variable in state : 
const [answerId, setAnswerId] = useState<boolean | undefined>();

Then I set it :
setAnswerId(false)

That I use inside a reaction
```
useDisposable(() => reaction(
        () => /*Random observable*/,
        (myObservable) => {
           console.log(answerId) //Always show undefined
        }
   }));
```

Shouldn't it works ?

**Comment by mweststrate** [comment]
No, useState doesn't create observables, so reactions won't react to them.
Instead, use `useEffect` instead if you want to react to useState created
things, or create some local observables, like: `const [store] =
useState(() => observable({ answerId: undefined }))`. See also #94

**Comment by MatthieuCoelho** [comment]
I'm agree with that, the trigger of the effect is a real observable for outside ...

Shouldn't the state be accessible inside the reaction ?

**Comment by mweststrate** [comment]
It is, but due to how javascript closures work, you will only see the first
value, which is probably what you are interested in, but technically correct

Op di 16 apr. 2019 01:12 schreef MatthieuC <notifications@github.com>:

**Comment by MatthieuCoelho** [comment]
@mweststrate thanks, so the only way to "fix" it is to use a local observable otherwise I won't me able to get the current state value.

**Comment by mweststrate** [comment]
There are several ways to fix it, such as using useEffect, instead of
useDisposable (which is going to be removed soon). However, we are still in
the progress of standardizing the _idiomatic_ ways to achieve this. See
also #97 and the open PR
